### ISSUE TITLE
MobileBert embedding vectors values

### ISSUE THREAD

**Issue by datapaintings** [huggingface/transformers#5540]
# 🐛 Bug

## Information

Model I am using (Bert, XLNet ...): MobileBert

Language I am using the model on (English, Chinese ...): English

The problem arises when using:
* [ ] the official example scripts: (give details below) 
* [ ] my own modified scripts: (give details below)

```py
def BERT_embeddings(data, max_seq_len):
    tokenizer = MobileBertTokenizer.from_pretrained('google/mobilebert-uncased')
    model = MobileBertModel.from_pretrained('google/mobilebert-uncased')
    batches = np.array_split(data, 5)  
    encoded = None
    for batch in batches:
        encoded_text = tokenizer.batch_encode_plus(batch.values, max_length=max_seq_len, pad_to_max_length=True, truncation=True)
        input_ids = tf.constant(encoded_text['input_ids'])#[None, :]
        #attention_masks = encoded_data_train['attention_mask']
        outputs = model(input_ids)
        if encoded == None:
            encoded = outputs[1]
        else:
            encoded = np.concatenate((encoded, outputs[1]), axis=0)

    encoded_test_x = pd.DataFrame(data=encoded, index=test_x.index)
    return encoded_test_x

print(BERT_embeddings(data, max_seq_len=18)
```

The tasks I am working on is:
* [ ] an official GLUE/SQUaD task: (give the name)
* [ ] my own task or dataset: (give details below)
I would like to create sentence embeddings with MobileBert
## To reproduce

1. 'text' is one column pandas object 
2. run above code 
3.

Embeddings have a form of vectors with large values like [-24888514.0  222398.593750...]

## Expected behavior
Should have [-1, 1] range

I would expect that elements of a vector will not be in range hundred of thousands but [-1, 1]

## Environment info

     
- `transformers` version: 3.0.1
- Platform:
- Python version: 3.7.3
- PyTorch version (GPU?): -
- Tensorflow version (GPU?): 2.0.0
- Using GPU in script?: No
- Using distributed or parallel set-up in script?: No

**Comment by LysandreJik** [comment]
Hi! Firstly, in your code you're using a `MobileBertModel` with tensorflow inputs; does that work? You should either use torch inputs or use `TFMobileBertModel` instead.

The model outputs have no reason to be bound between [-1, 1], as they're logits. You can pass them through a softmax layer to get the total sum to be 1 and have each result bound between -1 and 1!

**Comment by datapaintings** [comment]
Hello, thank you for your answer. I modified the function according to your suggestion, applying softmax on outputs to get hidden states values:

def softmax(x):
    """Compute softmax values for each sets of scores in x."""
    e_x = np.exp((x - np.max(x))/2000000)
    return e_x / e_x.sum(axis=0)

def BERT_embeddings2(test_x, max_seq_len):
    tokenizer = MobileBertTokenizer.from_pretrained('google/mobilebert-uncased')
    model = TFMobileBertModel.from_pretrained('google/mobilebert-uncased')
    batches = np.array_split(test_x, 100)  
    encoded = None
    for batch in batches:
        encoded_text = tokenizer.batch_encode_plus(batch.values, max_length=max_seq_len, pad_to_max_length=True, truncation=True)
        input_ids = tf.constant(encoded_text['input_ids'])#[None, :]
        outputs = model(input_ids)
        hidden_states = softmax(outputs[1])
        if encoded is None:
            encoded = hidden_states
        else:
            encoded = np.concatenate((encoded, hidden_states), axis=0)

Then I used encoded data to run experiment on multiple neural network architectures. What I observed, is that the loss function does not converge and is similar for various network shapes. This is not appearing while training on (not mobile) Bert Model. Do you know what can be the reason?
